quadriga/src/main/java/edu/asu/diging/quadriga/core/citesphere/impl/CitesphereConnectorImpl.java

```diff
@@ -31,12 +38,16 @@
 import com.nimbusds.oauth2.sdk.id.ClientID;
 import com.nimbusds.oauth2.sdk.token.AccessToken;
 
+import edu.asu.diging.quadriga.api.v1.model.TokenInfo;
 import edu.asu.diging.quadriga.core.citesphere.CitesphereConnector;
+import edu.asu.diging.quadriga.core.exceptions.OAuthException;
 import edu.asu.diging.quadriga.core.model.citesphere.CitesphereAppInfo;
+import edu.asu.diging.quadriga.core.model.citesphere.CitesphereToken;
+import edu.asu.diging.quadriga.core.model.citesphere.CitesphereUser;
 
 @Service
 @PropertySource("classpath:/config.properties")
-public class CitesphereConnectorImpl implements CitesphereConnector {
+public class CitesphereConnectorImpl implements CitesphereConnector, AuthenticationProvider {
```

Comment:
I would separate these two interfaces into separate classes. One is responsible to talk to Citesphere, the other to authenticate users.